src/sentry/middleware/ratelimit.py

```diff
@@ -23,20 +23,29 @@
 )
 
 
-class RatelimitMiddleware(MiddlewareMixin):
-    """Middleware that applies a rate limit to every endpoint."""
+class RatelimitMiddleware:
+    """Middleware that applies a rate limit to every endpoint.
+    See: https://docs.djangoproject.com/en/4.0/topics/http/middleware/#writing-your-own-middleware
+    """
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request: Request) -> Response:
+        # First, check if the endpoint call will violate.
+        data = {}
```

Comment:
what's the point of this `data` variable?

Reply:
I guess it doesn't need to be in a dictionary, but I was following up on a TODO left from before that said to put the fields in an object. I have to initialize it here to get around an unbound variable error.

Reply:
ah yes, that was at the time when we needed to attach it to the request object. This way all the things that were needed to be bubbled up to other places outside the middleware were all under one namespace and not scattered around. If you're not attaching these things to the request, then there's no reason to put them in an object.